presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -367,6 +376,24 @@ public void enforceMemoryLimits()
     }
 
     /**
+     * Enforce timeout at the query level
+     */
+    public void enforceQueryMaxRuntimeLimits()
+    {
+        Iterable<QueryExecution> runningQueries = queries.values().stream()
+                .filter(query -> query.getQueryInfo().getState() == RUNNING)
+                .collect(toImmutableList());
+
+        for (QueryExecution runningQuery : runningQueries) {
+            Duration queryMaxRuntime = SystemSessionProperties.getQueryMaxRuntime(runningQuery.getSession());
```

Comment:
if feels more natural to me to write

    if (query.getQueryInfo().getState() != RUNNING) {
        continue;
    }

at the beginning of the loop instead of preprocessing `queries.values()` with `stream()` prior to the loop.

What do you think, @cberner?